packages/flutter/test/material/scaffold_test.dart

```diff
@@ -389,6 +389,40 @@ void main() {
     expect(scrollable.position.pixels, equals(0.0));
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
 
+  testWidgets('Tapping the status bar scrolls to top use more appropriate curve', (WidgetTester tester) async {
+
+    // scrolls to top animation:
+    // Duration: 1000ms
+    // Curve: easeOutCirc
+
+    const int duration = 1000;
+    final List<double> stops = [0.842, 0.959, 0.993, 1.0];
+    const double scrollOffset = 1000;
+
+    await tester.pumpWidget(_buildStatusBarTestApp(debugDefaultTargetPlatformOverride));
+    final ScrollableState scrollable = tester.state(find.byType(Scrollable));
+    scrollable.position.jumpTo(scrollOffset);
+    await tester.tapAt(const Offset(100.0, 10.0));
+
+    await tester.pump(Duration.zero);
+    expect(scrollable.position.pixels, equals(scrollOffset));
+
+    for (int i = 0; i < stops.length; i++) {
+      await tester.pump( Duration(milliseconds: 1000 ~/ stops.length));
+      // scroll pixcels of this step is very long double, so compare with
+      // floored position pixcels
```

Comment:
Maybe something like:
```suggestion
      // Scroll pixel position is very long double, compare with
      // floored int pixel position.
```